### ISSUE TITLE
[FeatureRequest]Could we have a primitive to return binary data from a URL

### ISSUE THREAD

**Issue by cymplecy** [jmoenig/Snap#3172]
Just being cheeky but if you don't ask you don't get :)

I needed to access the binary data returned from a URL request and modified your Snap! code to get it

![ff2832b0617355ddc277283af5e09e04d3a46d93 (1)](https://user-images.githubusercontent.com/2357428/215272150-226f7141-c007-4e8b-a660-3e01cf70c9cf.png)

Would there be some chance of getting this (but properly coded of course) as a primitive or adding it as some sort of option to to the existing one?

**Comment by jmoenig** [comment]
nope, sorry, not as a general purpose primitive for the impending v8.1 release. Why don't you add it to your MQTT extension?

**Comment by cymplecy** [comment]
Its not an MQTT thing so I wouldn't want to pollute the library :)

**Comment by jmoenig** [comment]
Did you see the new `text to utf8` and `text from utf8` extension primitives? Don't those already do what you want?

**Comment by cymplecy** [comment]
Unfortunately not AFAIC - the standard URL block has already returned the data as UTF-16 (I think) so the data is already "corrupted" and the txt->utf8 isn't able to decode it correctly

The real binary returned by the my JS is

![TTS_textToSpeech_sound script pic (9)](https://user-images.githubusercontent.com/2357428/215325756-97a520ba-7c54-436e-8843-8e6d12759a16.png)

The data returned by the standard URL reporter is
![TTS_textToSpeech_sound script pic (10)](https://user-images.githubusercontent.com/2357428/215325798-026f8847-fdab-4ccf-b3ec-5f8f8d426143.png)

And AFAICT the txt->utf8 cannot recover the values at items 5,26 and 29
![TTS_textToSpeech_sound script pic (11)](https://user-images.githubusercontent.com/2357428/215325858-9ed65d53-dc6a-44ef-8aa9-0a1164e6067e.png)

**Comment by cycomachead** [comment]
I think a tool like this makes sense in the web services library personally. It would be useful - and I don’t remember if we’ve ever added a way to nicely grab images and sounds. -- Michael BallFrom my iPhonemichaelball.coOn Jan 29, 2023, at 4:26 AM, Simon Walters ***@***.***> wrote:﻿
Unfortunately not AFAIC - the standard URL block has already returned the data as UTF-16 (I think) so the data is already "corrupted" and the txt->utf8 isn't able to decode it correctly
The real binary returned by the my JS is

The data returned by the standard URL reporter is

And AFAICT the txt->utf8 cannot recover the values at items 5,26 and 29

**Comment by jmoenig** [comment]
yeah, the web services library would probably be a reasonable place to put this. We do have an extension primitive for importing a costume from a url since last summer, btw.

**Comment by cymplecy** [comment]
Just re-raising this (given all the discussion about importing large amounts of data)
